src/addressbook.rs

```diff
@@ -0,0 +1,116 @@
+//! # Addressbook utils
+
+use super::core::Address;
+use glob::glob;
+use serde_json;
+use std::fs::File;
+use std::fs::remove_file;
+use std::path::{Path, PathBuf};
+
+/// Addressbook Service
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct Addressbook {
+    dir: PathBuf,
+}
+
+/// Addressbook Errors
+#[derive(Debug, Clone)]
+pub enum AddressbookError {
+    /// IO Error
+    IO,
+
+    /// Invalid Address
+    InvalidAddress,
+}
+
+impl Addressbook {
+    /// Initialize new addressbook service for a dir
+    pub fn new(dir: PathBuf) -> Addressbook {
+        Addressbook { dir: dir }
+    }
+
+    fn read_json(path: &Path) -> Result<serde_json::Value, AddressbookError> {
+        match File::open(path) {
+            Ok(f) => serde_json::from_reader(f).or(Err(AddressbookError::IO)),
+            Err(_) => Err(AddressbookError::IO),
+        }
+    }
+
+    /// List all entries in the addressbook
+    pub fn list(&self) -> Vec<serde_json::Value> {
+        let files = glob(&format!("{}/*.json", &self.dir.to_str().unwrap())).unwrap();
```

Comment:
In production better to use `expect` method instead of `unwrap`, clippy rust linter show a warning for every `unwrap` call.